src/main.rs

```diff
@@ -227,21 +239,52 @@ fn main() {
             &camera.view_matrix(),
         );
 
+        #[cfg(debug_assertions)]
         ui::draw_fps_window(&imgui_ui);
 
+        // Clicking any of the models in the list means clearing everything out
+        // of the scene, importing given model and pushing it into scene.
+        if let Some(clicked_model) = ui::draw_model_window(&imgui_ui, &obj_filenames) {
+            let clicked_model_path = obj_file_paths
+                .get(&clicked_model)
+                .expect("Should get clicked model path from hash map");
+
+            match importer.import_obj(&clicked_model_path.to_str().unwrap()) {
+                Ok(models) => {
+                    for model in models {
+                        let geometry = Geometry::from_positions_and_normals_indexed(
+                            model.indices,
+                            model.vertex_positions,
+                            model.vertex_normals,
+                        );
+                        let model_handle =
+                            viewport_renderer.add_geometry(&device, &geometry).unwrap();
+
+                        for model in dynamic_models.iter() {
+                            viewport_renderer.remove_geometry(*model);
+                        }
+
+                        dynamic_models.clear();
```

Comment:
This is a leak. You need to `viewport_renderer.remove_geometry(id)` for each one you cleared. You can use the handy `Vec::drain()` for that (https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain)

Later we might think about adding a `Drop` impl to `RendererGeometryId` (or is still called `GeometryId` in master? 😂 ) that would remove itself from the renderer automatically, but may not be worth the trouble.

Reply:
My bad. Didn't notice it 🙏